### ISSUE TITLE
incorrect conv2d result

### ISSUE THREAD

**Issue by simochen** [pytorch/pytorch#24172]
## 🐛 Bug

The convolution between Tensor "a" and a kernel with all elements equal to 1 is supposed to be equal to a.sum(). The result is incorrect.

## To Reproduce

import torch
import torch.nn.functional as F
```
>>> a = torch.ones((1,1,62,62)) * 242 / 62 / 62
>>> weight = torch.ones((1,1,62,62))
>>> F.conv2d(a.pow(2), weight)
tensor([[[[15.2355]]]])
>>> a.pow(2).sum()
tensor(15.2352)
```
## Environment

 - PyTorch Version (e.g., 1.0): 1.1.0
 - OS (e.g., Linux): macOS
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): conda install pytorch torchvision -c pytorch
 - Python version: 3.7.3
 - CUDA/cuDNN version: None
 - GPU models and configuration:
 - Any other relevant information:

**Comment by tczhangzhi** [comment]
This is because the effective digit of float32 is only 7 digits. If you change to float64, it is the result you want.
```
>>> a = torch.ones((1,1,62,62), dtype=torch.float64) * 242 / 62 / 62
>>> weight = torch.ones((1,1,62,62), dtype=torch.float64)
>>> F.conv2d(a.pow(2), weight)
tensor([[[[15.2352]]]], dtype=torch.float64)
```

**Comment by simochen** [comment]
> This is because the effective digit of float32 is only 7 digits. If you change to float64, it is the result you want.
> 
> ```
> >>> a = torch.ones((1,1,62,62), dtype=torch.float64) * 242 / 62 / 62
> >>> weight = torch.ones((1,1,62,62), dtype=torch.float64)
> >>> F.conv2d(a.pow(2), weight)
> tensor([[[[15.2352]]]], dtype=torch.float64)
> ```

But I got correct result in pytorch 1.0.1.post2?

**Comment by ezyang** [comment]
@simochen The reason for this is probably that we changed the convolution algorithm being used, which results in the difference in numerical accuracy. The most likely change that happened in this time period was we turned on MKL DNN convolutions.

**Comment by soumith** [comment]
aso others pointed out, this is reasonable outcome for float32 precision

**Comment by tczhangzhi** [comment]
> > This is because the effective digit of float32 is only 7 digits. If you change to float64, it is the result you want.
> > ```
> > >>> a = torch.ones((1,1,62,62), dtype=torch.float64) * 242 / 62 / 62
> > >>> weight = torch.ones((1,1,62,62), dtype=torch.float64)
> > >>> F.conv2d(a.pow(2), weight)
> > tensor([[[[15.2352]]]], dtype=torch.float64)
> > ```
> 
> But I got correct result in pytorch 1.0.1.post2?

It works well for me. Did you forget your `import`? :P
```
import torch
import torch.nn.functional as F
```
